query/language/pipelines.feature

```diff
@@ -0,0 +1,202 @@
+    Given typeql write query
+    """
+    insert
+      $p1 isa person, has name "Alice", has age 10, has ref 0;
+      $p2 isa person, has name "Bob", has age 11, has ref 1;
+    """
+    Given transaction commits
+
+    Given connection open write transaction for database: typedb
+    Given get answers of typeql write query
+    """
+    match
+      $p1 isa person, has name "Alice";
+      $p2 isa person, has name "Bob";
+    insert
+      $f isa friendship, links (friend: $p1, friend: $p2), has ref 3;
+    """
+    Then uniquely identify answer concepts
+      | f         | p1              | p2           |
+      | key:ref:3 | key:ref:0       | key:ref:1    |
+
+
+  Scenario: The set of variables being streamed can be reduced using select.
+    Given connection open write transaction for database: typedb
+    Given typeql write query
+    """
+    insert
+      $p1 isa person, has name "Alice", has age 10, has ref 0;
+    """
+    Given transaction commits
+
+    Given connection open read transaction for database: typedb
+    Given get answers of typeql read query
+    """
+    match
+      $p isa person, has name "Alice";
+    select $p;
+    """
+    Then uniquely identify answer concepts
+      | p              |
+      | key:ref:0      |
+
+  Scenario: A match can be chained with a delete and insert to emulate an update.
+    Given connection open write transaction for database: typedb
+    Given typeql write query
+    """
+    insert
+      $p1 isa person, has name "Alice", has age 10, has ref 0;
+    """
+    Given transaction commits
+
+    Given connection open write transaction for database: typedb
+    Given typeql write query
+    """
+    match
+      $p1 isa person, has name "Alice", has age $age;
+    delete
+      has $age of $p1;
+    insert
+      $p1 has age 11;
+    """
+    Given transaction commits
+    Given connection open read transaction for database: typedb
+    Given get answers of typeql read query
+    """
+    match
+      $p isa person, has name $name, has age $age;
+    """
+    Then uniquely identify answer concepts
+      | p         | name            | age          |
+      | key:ref:0 | attr:name:Alice | attr:age:11  |
+
+
+  Scenario: A stream can be sorted on a set of variables
+    Given connection open write transaction for database: typedb
+    Given typeql write query
+    """
+    insert
+      $p1 isa person, has name "Alice", has age 10, has ref 0;
+      $p2 isa person, has name "Bob", has age 11, has ref 1;
+    """
+    Given transaction commits
+
+    Given connection open read transaction for database: typedb
+    Given get answers of typeql read query
+    """
+    match
+      $p isa person, has name $name;
+    sort $name;
+    limit 1;
+    """
+    Then uniquely identify answer concepts
+      | p              |
+      | key:ref:0      |
+
+    Given get answers of typeql read query
+    """
+    match
+      $p isa person, has name $name;
+    sort $name desc;
```

Comment:
Maybe you should extend this with one that sorts on name, then something else? The test says "set of variables"?

Reply:
Renamed and moved the multi-variable case below